internal/otelcollector/config/metric/agent/set_instrumention_scope.go

```diff
@@ -0,0 +1,41 @@
+package agent
+
+import (
+	"fmt"
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config"
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config/metric"
+)
+
+func transformInstrumentationScope(inputSource metric.InputSourceType) *TransformProcessor {
+	return &TransformProcessor{
+		ErrorMode:        "ignore",
+		MetricStatements: makeMetricStatement(inputSource),
+	}
+}
+
+func makeMetricStatement(inputSource metric.InputSourceType) []config.TransformProcessorStatements {
+	var metricStatements []string
+
+	return []config.TransformProcessorStatements{
+		{
+			Context:    "scope",
+			Statements: append(metricStatements, makeInstrumentationStatement(inputSource)),
+		},
+	}
+}
+
+func makeInstrumentationStatement(inputSource metric.InputSourceType) string {
+	var instrumentationScopeName = map[metric.InputSourceType]string{
+		metric.InputSourceRuntime:    "otelcol/kubeletstatsreceiver",
+		metric.InputSourcePrometheus: "otelcol/prometheusreceiver",
+		metric.InputSourceIstio:      "otelcol/prometheusreceiver",
+	}
+
+	var transformedInstrumentationScopeName = map[metric.InputSourceType]string{
+		metric.InputSourceRuntime:    "io.kyma-project.telemetry/kubeletstatsreceiver",
```

Comment:
why kubeletstatsreceiver? The goal is to use the input name!

Reply:
it has been changed